rmf_gazebo_plugins/src/readonly.cpp

```diff
@@ -0,0 +1,466 @@
+{
+  // We do initialization only during ::Load
+}
+
+ReadonlyPlugin::~ReadonlyPlugin()
+{
+}
+
+rclcpp::Logger ReadonlyPlugin::logger()
+{
+  return rclcpp::get_logger("read_only_" + _model->GetName());
+}
+
+void ReadonlyPlugin::Load(gazebo::physics::ModelPtr model, sdf::ElementPtr sdf)
+{
+  _current_mode.mode = rmf_fleet_msgs::msg::RobotMode::MODE_MOVING;
+  _model = model;
+  _ros_node = gazebo_ros::Node::Get(sdf);
+
+  // Getting sdf elements
+  if (sdf->HasElement("level_name"))
+    _level_name = sdf->Get<std::string>("level_name");
+  RCLCPP_INFO(logger(), "Setting level name to: " + _level_name);
+
+  if (sdf->HasElement("graph_index"))
+    _nav_graph_index = sdf->Get<std::size_t>("graph_index");
+  RCLCPP_INFO(logger(), "Setting nav graph index: " + _nav_graph_index);
+
+  if (sdf->HasElement("spawn_waypoint"))
+    _start_wp_name = sdf->Get<std::string>("spawn_waypoint");
+  RCLCPP_INFO(logger(), "Setting start wp name: " + _start_wp_name);
+
+  if (sdf->HasElement("look_ahead"))
+    _lookahead = sdf->Get<std::size_t>("look_ahead");
+  _lookahead = _lookahead < 1 ? 1 : _lookahead;
+  _next_wp.resize(_lookahead);
+  _path.resize(_lookahead);
+  RCLCPP_INFO(logger(), "Setting lookahead: " + std::to_string(_lookahead));
+
+  if (sdf->HasElement("nominal_drive_speed"))
+    _v_n = sdf->Get<double>("nominal_drive_speed");
+  RCLCPP_INFO(logger(), "Setting nominal drive speed to: " + std::to_string(_v_n));
+
+  if (sdf->HasElement("nominal_drive_acceleration"))
+    _a_n = sdf->Get<double>("nominal_drive_acceleration");
+  RCLCPP_INFO(logger(), "Setting nominal drive acceleration to: " + std::to_string(_a_n));
+
+  if (sdf->HasElement("nominal_turn_speed"))
+    _w_n = sdf->Get<double>("nominal_turn_speed");
+  RCLCPP_INFO(logger(), "Setting nominal turn speed to:" + std::to_string(_w_n));
+
+  if (sdf->HasElement("nominal_turn_acceleration"))
+    _alpha_n = sdf->Get<double>("nominal_turn_acceleration");
+  RCLCPP_INFO(logger(), "Setting nominal turn acceleration to:" + std::to_string(_alpha_n));
+
+  RCLCPP_INFO(logger(), "hello i am " + model->GetName());
+
+  _update_connection = gazebo::event::Events::ConnectWorldUpdateBegin(
+      std::bind(&ReadonlyPlugin::OnUpdate, this));
+
+  robot_state_pub = _ros_node->create_publisher<rmf_fleet_msgs::msg::RobotState>(
+      "/robot_state", 10);
+
+  // Subscription to /map
+  auto qos_profile = rclcpp::QoS(10);
+  qos_profile.transient_local();
+  _building_map_sub = _ros_node->create_subscription<BuildingMap>(
+        "/map",
+        qos_profile,
+        std::bind(&ReadonlyPlugin::map_cb, this, std::placeholders::_1));
+
+  _current_task_id = "demo";
+  if (model->GetName().c_str())
+  _name = _model->GetName().c_str();
+  RCLCPP_INFO(logger(), "Setting name to: " + _name);
+
+}
+
+void ReadonlyPlugin::map_cb(const BuildingMap::SharedPtr msg)
+{
+  if (msg->levels.empty())
+  {
+    RCLCPP_ERROR(logger(), "Received empty building map");
+    return;
+  }
+
+  _found_map = true;
```

Comment:
this member doesn't seem to be used anywhere else, just putting a comment here in case you forgot to do something with it.
was it for a timeout?

Reply:
This variable is no longer used. I'll remove it.